### DISCUSSION TITLE
Understanding response.get()

### DISCUSSION THREAD

**Discussion by dekra54** [eModbus/eModbus#300] [Q&A] [answered]
Good Evening,

I have a "simple" question regarding the response.get() funtion.
Assuming i want to read an I32 (Signed Integer) from a ModBus TCP device.
Can i expect that the following simplified code "converts" the response automatically into an int32?
```c++
  int32_t current_power_value;   

  response.get(3,current_power_value);
```

**Comment by Miq1** [comment, marked as answer]
Yes, but only if the byte sequence in the message is MSB first. The bytes will be taken one by one and shifted into the ``int32``.

**Reply by dekra54** [reply]
Thank you very much for that fast reply :) . That answers my question.

Maybe it would be a good idea to add an example for these supported type conversions. 

Or if possible add something like
```c++
float responseData = response.getData.ToFloat();
```
which maybe  ,in my way of thinking, makes it a bit more intuitive.

**Reply by Miq1** [reply]
Well, there **are** ``get()`` varieties for ``float`` and ``double`` in fact that additionally allow you to do byte/register/word swapping if required. 🤣 
You may want to read the [docs](https://emodbus.github.io/reading-from-a-modbusmessage#uint16_t-getuint16_t-index-t-value)   ... 😉

**Reply by dekra54** [reply]
Yes now that you confirmed its kind of obvious reading the docs again 😄 
Thank you for the support and the great work with the library
